### PR TITLE
[PHPSpec to PHPUnit] Channel Bundle

### PR DESCRIPTION
**Pull request by rust-le**
| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | OP-571
| License         | MIT

Rewrite PHPSpec tests to PHPUnit in Channel Bundle

## Summary by CodeRabbit

- **Tests**
  - Added new PHPUnit test classes for FakeChannelCodeProvider, FakeChannelContext, and FakeChannelPersister to verify their behaviors and edge cases.
  - Removed PhpSpec specification files for these components, fully replacing them with PHPUnit-based tests.
- **Chores**
  - Updated PHPUnit configuration to use the latest schema and adjusted the test suite naming.
  - Updated composer configuration by removing an unused autoload-dev mapping.

### CODE REVIEW

File: src/Sylius/Bundle/ChannelBundle/phpunit.xml.dist [hunk 1]
```diff
@@ -1,15 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/8.5/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/phpunit-10.xsd"
```
Comment (Prometee) [inline src/Sylius/Bundle/ChannelBundle/phpunit.xml.dist]: Can you use the vendor reference instead of changing the version?

**Review by Prometee — CHANGES_REQUESTED** [review]

File: src/Sylius/Bundle/ChannelBundle/phpunit.xml.dist (hunk 1 above)
Comment (rust-le) [reply src/Sylius/Bundle/ChannelBundle/phpunit.xml.dist]: Thanks!

File: src/Sylius/Bundle/ChannelBundle/composer.json [hunk 2]
```diff
@@ -64,7 +64,6 @@
     },
     "autoload-dev": {
         "psr-4": {
-            "Sylius\\Bundle\\ChannelBundle\\spec\\": "spec/",
             "Sylius\\Bundle\\ChannelBundle\\Tests\\": "Tests/"
```
Comment (GSadee) [inline src/Sylius/Bundle/ChannelBundle/composer.json:67]: You could change it to be consistent with other packages:
```suggestion
            "Tests\\Sylius\\Bundle\\ChannelBundle\\": "Tests/"
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you @rust-le! 🎉
